controllers/elbv2/targetgroupbinding_controller.go

```diff
@@ -110,22 +114,26 @@ func (r *targetGroupBindingReconciler) reconcile(ctx context.Context, req reconc
 }
 
 func (r *targetGroupBindingReconciler) reconcileTargetGroupBinding(ctx context.Context, tgb *elbv2api.TargetGroupBinding) error {
+	defer r.metricsCollector.ObserveControllerReconcileLatency("targetGroupBinding", "add finalizers")
 	if err := r.finalizerManager.AddFinalizers(ctx, tgb, targetGroupBindingFinalizer); err != nil {
 		r.eventRecorder.Event(tgb, corev1.EventTypeWarning, k8s.TargetGroupBindingEventReasonFailedAddFinalizer, fmt.Sprintf("Failed add finalizer due to %v", err))
 		return err
 	}
 
+	defer r.metricsCollector.ObserveControllerReconcileLatency("targetGroupBinding", "reconcile")
```

Comment:
this seems not work due to how defer works in golang. 
ideally we should collect latency metrics spend for each task, instead of "latency spend since X stage".
There is a difference between
* latency spent on adding finalizers
* latency since adding finalizer to finish reconcile.
The first one is clearly the better option and allows us to monitor on